### PR TITLE
Add new config syntax for table filters.

### LINKED ISSUE #5779: Row-level security for MaterializedView
There is no possible way to configure row-level security filters (https://clickhouse.yandex/docs/en/operations/settings/settings_users/#user-name-databases) to MaterializedView (https://clickhouse.yandex/docs/en/operations/table_engines/materializedview/) because it is actually reads data from .inner.* table, but according to XML spec (https://www.w3.org/TR/REC-xml/#sec-common-syn) is not possible to start a node name from ".".

This XML is incorrect:
```
<databases>
    <my_database>
        <.inner.my_view>
            <filter>number = 42</filter>
        </.inner.my_view>
    </my_database>
</databases>
```

Got an error like this:
> 2019.06.27 17:45:23.764880 [ 1 ] {} <Error> Application: SAXParseException: Invalid token in '/etc/clickhouse-server/users.xml', line 52 column 21

ClickHouse version == 19.9.2.4

### PR DESCRIPTION
**Pull request by abyss7**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):
Support new syntax for row-level security filters: `<table name='table_name'>…</table>`. Fixes #5779

### CODE REVIEW

File: dbms/src/Interpreters/Users.cpp [hunk 1]
```diff
@@ -120,11 +121,24 @@ User::User(const String & name_, const String & config_elem, const Poco::Util::A
             /// Read table properties
             for (const auto & table : table_names)
             {
-                const auto config_filter = config_database + "." + table + ".filter";
+                String table_name = table;
+                String config_filter = config_database + "." + table_name + ".filter";
+
+                /// FIXME: naming scheme can be spoiled by user.
+                if (table.rfind("table[", 0) == 0)
```
Comment (alexey-milovidov) [inline dbms/src/Interpreters/Users.cpp]: Don't understand, why `rfind` instead of `find` and what if there is only one <table> element and how can it be spoiled by user?

Comment (abyss7) [reply dbms/src/Interpreters/Users.cpp]: Will add a test to check a single element.
The comment about spoiler is obsolete - will remove it.
`rfind` is just copy-paste - there is no particular reason.

File: dbms/programs/server/users.xml [hunk 2]
```diff
@@ -83,30 +83,7 @@
 
             <!-- Quota for user. -->
             <quota>default</quota>
-
```
Comment (vitlibar) [inline dbms/programs/server/users.xml]: Why did you delete these comments? They seem useful.

File: dbms/src/Access/UsersConfigAccessStorage.cpp [hunk 3]
```diff
@@ -139,9 +139,21 @@ namespace
                     config.keys(database_config, table_names);
 
                     /// Read table properties
-                    for (const String & table_name : table_names)
+                    for (const String & table : table_names)
                     {
-                        const auto filter_config = database_config + "." + table_name + ".filter";
+                        String table_name = table;
```
Comment (vitlibar) [inline dbms/src/Access/UsersConfigAccessStorage.cpp]: Maybe it would be better to rename `table` to something like `key_in_database_config`. Currently you have two variables `table` and `table_name` in the same scope and it looks confusive.

File: dbms/tests/integration/test_row_policy/single_table_name.xml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0"?>
```
Comment (vitlibar) [inline dbms/tests/integration/test_row_policy/single_table_name.xml]: It doesn't seem to be easy to understand the names of these test xml files: `custom` doesn't really mean "custom" because all names are "custom" in some way, and `single` doesn't really mean "single", because there are three tables there.

I think it's better to rename these files to make the tests clearer, e.g.
`single_table_name.xml` -> `tag_with_table_name.xml`,
`custom_table_name.xml` -> `multiple_tags_with_table_names.xml`.

File: dbms/tests/integration/test_row_policy/test.py [hunk 5]
```diff
@@ -28,13 +28,19 @@ def started_cluster():
...
+            INSERT INTO mydb.table values (0, 0), (0, 1), (1, 0), (1, 1);
+
             CREATE TABLE mydb.filtered_table2 (a UInt8, b UInt8, c UInt8, d UInt8) ENGINE MergeTree ORDER BY a;
             INSERT INTO mydb.filtered_table2 values (0, 0, 0, 0), (1, 2, 3, 4), (4, 3, 2, 1), (0, 0, 6, 0);
 
             CREATE TABLE mydb.filtered_table3 (a UInt8, b UInt8, c UInt16 ALIAS a + b) ENGINE MergeTree ORDER BY a;
             INSERT INTO mydb.filtered_table3 values (0, 0), (0, 1), (1, 0), (1, 1);
+
+            CREATE TABLE mydb.`.filtered_table3` (a UInt8, b UInt8, c UInt16 ALIAS a + b) ENGINE MergeTree ORDER BY a;
+            INSERT INTO mydb.`.filtered_table3` values (0, 0), (0, 1), (1, 0), (1, 1);
```
Comment (vitlibar) [inline dbms/tests/integration/test_row_policy/test.py]: `.filtered_table3` looks like `filtered_table3`. The name `.filtered_table4.` seems less confusive.

File: dbms/programs/server/users.xml (hunk 2 above)
Comment (abyss7) [reply dbms/programs/server/users.xml]: Because it's just comments - if we want provide an example, then we should do it for all possible options. Otherwise, it's good to put an example into documentation.

File: dbms/src/Access/UsersConfigAccessStorage.cpp [hunk 6]
```diff
@@ -139,9 +139,21 @@ namespace
...
 
                     /// Read table properties
-                    for (const String & table_name : table_names)
+                    for (const String & key_in_database_config : table_names)
                     {
-                        const auto filter_config = database_config + "." + table_name + ".filter";
+                        String table_name = key_in_database_config;
+                        String filter_config = database_config + "." + table_name + ".filter";
+
+                        if (key_in_database_config.find("table[") == 0)
```
Comment (vitlibar) [inline dbms/src/Access/UsersConfigAccessStorage.cpp]: `key_in_database_config.starts_with("table[")` looks more simple.

File: dbms/src/Access/UsersConfigAccessStorage.cpp [hunk 7]
```diff
@@ -139,9 +139,21 @@ namespace
                     config.keys(database_config, table_names);
 
                     /// Read table properties
-                    for (const String & table_name : table_names)
+                    for (const String & key_in_database_config : table_names)
```
Comment (vitlibar) [inline dbms/src/Access/UsersConfigAccessStorage.cpp]: The name `keys_in_database_config` instead of `table_names` seems more consistent here.

File: dbms/programs/server/users.xml (hunk 2 above)
Comment (vitlibar) [reply dbms/programs/server/users.xml]: It seems we have no documentation about row level security yet. So these comments were only documentation about that.
